Poker.BE/Poker.BE.Service/Services/AuthenticationService.cs

```diff
@@ -1,13 +1,91 @@
-﻿using System;
+﻿using Poker.BE.Service.Modules.Results;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using Poker.BE.Service.Modules.Requests;
+using Poker.BE.Domain.Core;
+using Poker.BE.Domain.Utility.Exceptions;
+using Poker.BE.Domain.Security;
+using Poker.BE.Domain.Utility.Logger;
 
 namespace Poker.BE.Service.Services
 {
-    public class AuthenticationService : IServices.IAuthenticationService
-    {
+	public class AuthenticationService : IServices.IAuthenticationService
+	{
+		#region fields
+		public UserManager Manager;
+		#endregion
 
-    }
+		#region properties
+		public IDictionary<int, User> Users { get; set; }
+		#endregion
+
+		public LoginResult Login(LoginRequest request)
+		{
+			var result = new LoginResult();
+			try
+			{
+				result.User = Manager.LogIn(request.UserName, request.Password).GetHashCode();
+			}
+			catch (UserNotFoundException e)
+			{
+				result.Success = false;
+				result.ErrorMessage = e.Message;
+			}
+			catch (IncorrectPasswordException e)
+			{
+				result.Success = false;
+				result.ErrorMessage = e.Message;
+			}
+			return result;
+		}
+		public LogoutResult Logout(LogoutRequest request)
+		{
+			var result = default(LogoutResult);
+			User user;
+			if (!Users.TryGetValue(request.User, out user))
+			{
+				result.Success = false;
+				result.ErrorMessage = "User ID {0} not found";
+			}
+			try
+			{
+				result.User = Manager.LogOut(user).GetHashCode();
+			}
+			catch (UserNotFoundException e)
+			{
+				result.Success = false;
+				result.ErrorMessage = e.Message;
+			}
+			return result;
+		}
+		public SignUpResult SignUp(SignUpRequest request)
```

Comment:
👎 please keep space between functions - your code is to dense - it's hard to understand that way.